frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/lambda/CheckReAuthUserHandlerTest.java

```diff
@@ -355,23 +349,43 @@ void shouldReturn400WhenUserHasBeenBlockedForPasswordRetries(
     void shouldReturn404ForWhenUserDoesNotMatch(boolean supportAuthenticationAttempts) {
         when(configurationService.isAuthenticationAttemptsServiceEnabled())
                 .thenReturn(supportAuthenticationAttempts);
+        when(clientRegistry.getRedirectUrls()).thenReturn(List.of(INTERNAL_SECTOR_URI));
+        when(userContext.getTxmaAuditEncoded()).thenReturn(null);
+        when(codeStorageService.getIncorrectEmailCount(EMAIL_USED_TO_SIGN_IN)).thenReturn(1);
+        when(authenticationAttemptsService.getCount(
+                        TEST_SUBJECT_ID, JourneyType.REAUTHENTICATION, CountType.ENTER_EMAIL))
+                .thenReturn(1);
+
+        var userProfile = generateUserProfile();
+
+        var expectedRpPairwiseSub =
+                ClientSubjectHelper.getSubject(
+                                userProfile,
+                                clientRegistry,
+                                authenticationService,
+                                INTERNAL_SECTOR_URI)
+                        .getValue();
+
         var event = apiRequestEventWithHeadersAndBody(VALID_HEADERS, null);
 
         var result =
                 handler.handleRequestWithUserContext(
                         event,
                         context,
                         new CheckReauthUserRequest(
-                                DIFFERENT_EMAIL_USED_TO_REAUTHENTICATE, TEST_RP_PAIRWISE_ID),
+                                DIFFERENT_EMAIL_USED_TO_REAUTHENTICATE, expectedRpPairwiseSub),
                         userContext);
 
         assertEquals(404, result.getStatusCode());
         assertThat(result, hasJsonBody(ErrorResponse.ERROR_1056));
 
         verify(auditService)
                 .submitAuditEvent(
-                        FrontendAuditableEvent.AUTH_REAUTHENTICATION_INVALID,
-                        testAuditContextWithAuditEncoded);
+                        FrontendAuditableEvent.AUTH_REAUTH_INCORRECT_EMAIL_ENTERED,
+                        testAuditContextWithoutAuditEncoded,
```

Comment:
I think here we're also missing some other fields (from the restricted section and the user) that is in the schema PR